python-scripts/eeprom-util.py

```diff
@@ -0,0 +1,200 @@
+  print ('Encrypted Game Region Code:\t{}'.format(''.join(format(x, '02x') for x in eeprom.gameReg)))
+
+  print ('\nIDENTITY EEPROM DATA')
+  print ('Identity Checksum:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.identChk)))
+  print ('Serial Number:\t\t\t{}'.format(eeprom.serial.decode('ascii')))
+  print ('MAC Address:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.mac)))
+  print ('Online Key:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.onlineKey)))
+
+  # print NTSC / PAL instead of their respective codes
+  if int.from_bytes(eeprom.videoStd, 'big', signed=False) == 81920: # 81920 == 00014000 
+    print ('Video Standard:\t\t\tNTSC ({})'.format(''.join(format(x, '02x') for x in eeprom.videoStd)))  
+  elif int.from_bytes(eeprom.videoStd, 'big', signed=False) == 229376: # 229376 == 00038000
+    print ('Video Standard:\t\t\tPAL ({})'.format(''.join(format(x, '02x') for x in eeprom.videoStd)))
+  else:
+    print ('Video Standard:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.videoStd)))
+
+  if args.keyFileLoc:
+    # load eeprom key
+    eepKeyFile = open(args.keyFileLoc, 'rb')
+    eepKey = eepKeyFile.read()
+    eepKeyFile.close()
+  
+    # generate hmac-sha1 hash of eeprom key
+    eepKeyHash = hmac.new(eepKey, eeprom.hash, sha1).digest()
+
+    # initialize EepKeyObj method object
+    eepKeyObj = EepKeyObj()
+    eepKeyPrepCrypt(eepKeyHash, eepKeyObj)
+  
+    # decrypt confounder 
+    decConf = eepKeyCrypt(eepDump[20:28], eepKeyObj)
+
+    # decrypt HDD key and game region code
+    decData = eepKeyCrypt(eepDump[28:48], eepKeyObj)
+
+    # we need to hash the confounder and data as one
+    hashData = decConf + decData
+
+    # hash decrypted EEPROM data
+    eepHash = hmac.new(eepKey, hashData, sha1).digest()
+
+    # make sure hashes match
+    if eepHash == eeprom.hash:
+      print ('\nDECRYPTED EEPROM DATA')
+
+      # pull HDD key from decryted data
+      decHddKey = bytearray()
+      for cnt in range(16):
+        decHddKey.append(decData[cnt])
+
+      # pull game region from decrypted data
+      decGameReg = bytearray()
+      for cnt in range(4):
+        decGameReg.append(decData[cnt + 16])
+
+      print ('Verified EEPROM Hash:\t\t{}'.format(''.join(format(x, '02x') for x in eepHash)))      
+      print ('Decrypted Confounder:\t\t{}'.format(''.join(format(x, '02x') for x in decConf)))
+      print ('Decrypted HDD Key:\t\t{}'.format(''.join(format(x, '02x') for x in decHddKey)))
+      print ('Decrypted Game Region Code:\t{}'.format(''.join(format(x, '02x') for x in decGameReg)))
+
+    # hashes don't match
+    else:
+      print ('\nWARNING: The decrypted and encrypted EEPROM hashes do not match!')
+      print ('Either your EEPROM file or EEPROM key is corrupt / invalid!\n')
+      print ('Original EEPROM Hash:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.hash)))
+      print ('Invalid EEPROM Hash:\t\t{}'.format(''.join(format(x, '02x') for x in eepHash)))
```

Comment:
nit: A hash mismatch should not prevent us from outputting additional data (such as the keys) in my opinion.
As long as it's obvious that there was a mismatch (and which data it applies to / which data might be broken or wrong) we should give as much information as possible.

Reply:
If the has matches, the key was wrong, meaning the whole decryption was wrong. It would be useless information to know all the wrong contents?